coffea/hist/hist_tools.py

```diff
@@ -963,14 +963,28 @@ def fill(self, **values):
                 self._sumw2[sparse_key] = np.zeros(shape=self._dense_shape, dtype=self._dtype)
 
         if self.dense_dim() > 0:
-            dense_indices = tuple(d.index(values[d.name]) for d in self._axes if isinstance(d, DenseAxis))
+            dense_indices = ()
+            for d in self._axes:
+                if isinstance(d, DenseAxis):
+                    if not isinstance(values[d.name], np.ndarray):
```

Comment:
You could consider instead to try `numpy.asarray(values[d.name])` to give a chance for compatible arrays to be cast to numpy. What is not clear to me is where integer arguments are rejected. What numpy function didn't accept them?

Reply:
I was about to make the same comment. We should preserve original behavior, and it doesn't look like it is that hard.

Reply:
It's np.bincount - it expects array of dimension 1 exactly (and single numbers are dim 0).

Reply:
Ah, so the solution is to wrap the `xy` indexer with `np.atleast_1d`

Reply:
Sure, good idea :) I can turn `values[d.name]` into a 1d array in case it is a list or a single digit with `np.atleast_1d`. I guess I need to add then a check, that it returns an array of dimension 1 exactly?